app/services/catalog/update_order_item.rb

```diff
@@ -46,13 +46,23 @@ def fetch_external_url
     def mark_item_based_on_status
       case @payload["status"]
       when "ok"
-        mark_item_finished
-        @order_item.order.finalize_order
+        case @payload["state"]
+        when "completed"
+          mark_item_finished
+          @order_item.order.finalize_order
+        when "running"
+          @order_item.state = "Processing"
```

Comment:
@eclarizio Do we need to set the state?

Reply:
@mkanoor I don't know, I just kinda figured we did, since as far as I understand it starts as "Created", and then before this PR would either get directly moved to "Completed" or "Failed". If you don't think having a state for "Processing" is useful, I can remove it. Your call.

Reply:
@eclarizio Lets remove it, since we can have Processing from Approval also and it would be confusing. Since these progress messages come in order it would be pretty apparent who sent the message